### PR TITLE
Switch to Ace editor

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/v3-project-management/issues/215

@iHiD Have a play with this one. I think the way we have abstracted our file editor makes it easy for us to switch between Ace and Monaco. I've finished this in about an hour, and I don't think we will run into problems if we decide to use Ace editor in the future.

In terms of the features that both editors bring, I can't really tell the difference. Maybe we could discuss later.

### CODE REVIEW

**Comment by iHiD** [comment]
@kntsoriano Let's do this! It's good. Could you get it ready to merge pls (Note I rebased and added a commit)

**Comment by kntsoriano** [comment]
Screen Shot 2021-04-22 at 3 53 24 PM

Do we still need this button? Previously, we opened the Monaco command palette when this button was clicked. Ace doesn't have a command palette.

File: app/javascript/components/editor/FileEditorAce.tsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  editorDidMount: (editor: FileEditorHandle) => void
+  language: string
+  onRunTests: () => void
+  onSubmit: () => void
+  theme: string
+  keybindings: Keybindings
+  files: File[]
+  wrap: WrapSetting
+}): JSX.Element {
+  /* TODO: These should be set from the track config */
```
Comment (kntsoriano) [inline app/javascript/components/editor/FileEditorAce.tsx:39]: @iHiD How do I get this config?

**Comment by kntsoriano** [comment]
@iHiD Should be ready :)

**Comment by iHiD** [comment]
@kntsoriano Tests are failing.

**Comment by kntsoriano** [comment]
@iHiD Should be good now :)

**Comment by iHiD** [comment]
Nice work @kntsoriano :)

**Comment by ErikSchierboom** [comment]
> I've finished this in about an hour

That's seriously impressive @kntsoriano!

**Pull request merged**